lib/email/sender.rb

```diff
@@ -226,6 +226,23 @@ def send
       end
 
       MessageBuilder.custom_headers(SiteSetting.email_custom_headers).each do |key, _|
+        # Any custom headers added via MessageBuilder that are doubled up here
+        # with values that we determine should be set to the last value, which is
+        # the one we determined. Our header values should always override the email_custom_headers.
+        #
+        # While it is valid via RFC5322 to have more than one value for certain headers,
+        # we just want to keep it to one, especially in cases where the custom value
+        # would conflict with our own.
+        #
+        # See https://datatracker.ietf.org/doc/html/rfc5322#section-3.6 and
+        # https://www.rubydoc.info/github/mikel/mail/Mail/Header
```

Comment:
I would just link directly to the gem here instead of relying on `rubydoc.info`

Reply:
Ah yep fair enough